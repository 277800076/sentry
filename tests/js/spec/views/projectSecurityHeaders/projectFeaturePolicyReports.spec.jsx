import React from 'react';

import {shallow} from 'enzyme';
import ProjectFeaturePolicyReports from 'app/views/settings/projectSecurityHeaders/featurePolicy';

describe('ProjectFeaturePolicyReports', function() {
  let org = TestStubs.Organization();
  let project = TestStubs.Project();
  let url = `/projects/${org.slug}/${project.slug}/feature-policy/`;

  beforeEach(function() {
    MockApiClient.clearMockResponses();
    MockApiClient.addMockResponse({
      url: `/projects/${org.slug}/${project.slug}/keys/`,
      method: 'GET',
      body: [],
    });
  });

  it('renders', function() {
    let wrapper = shallow(
      <ProjectFeaturePolicyReports
        organization={org}
        project={project}
        setProjectNavSection={() => {}}
        {...TestStubs.routerProps({
          params: {orgId: org.slug, projectId: project.slug},
          location: TestStubs.location({pathname: url}),
        })}
      />,
      TestStubs.routerContext()
    );
    expect(wrapper).toMatchSnapshot();
  });
});
