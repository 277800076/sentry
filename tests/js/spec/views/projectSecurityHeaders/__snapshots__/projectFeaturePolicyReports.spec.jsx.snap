// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFeaturePolicyReports renders 1`] = `
<SideEffect(DocumentTitle)
  title="Sentry"
>
  <div>
    <SettingsPageHeading
      title="Feature Policy"
    />
    <PreviewFeature />
    <ReportUri
      keyList={Array []}
      params={
        Object {
          "orgId": "org-slug",
          "projectId": "project-slug",
        }
      }
    />
    <Panel>
      <PanelHeader>
        About
      </PanelHeader>
      <PanelBody
        direction="column"
        disablePadding={false}
        flex={false}
      >
        <p>
          <span
            key="4"
          >
            <ExternalLink
              href="https://github.com/WICG/feature-policy"
              key="1"
              rel="noreferrer noopener"
              target="_blank"
            >
              <span
                key="0"
              >
                Feature Policy
              </span>
            </ExternalLink>
            <span
              key="2"
            >
              
      is a security standard which gives a website the ability to allow and deny the use of browser features in its own frame, and in iframes that it embeds
            </span>
          </span>
        </p>
        <p>
          <span
            key="4"
          >
            <span
              key="0"
            >
              To configure reports in Sentry, you'll need to configure the 
            </span>
            <code
              key="1"
            >
              Report-To
            </code>
            <span
              key="2"
            >
               a header from your server:
            </span>
          </span>
        </p>
        <pre>
          Report-To: {"url": "https://sentry.example.com/api/security-report/", "max-age": 86400}
        </pre>
        <p>
          <span
            key="5"
          >
            <span
              key="0"
            >
              For more information, see 
            </span>
            <a
              href="https://github.com/WICG/feature-policy/blob/master/reporting.md"
              key="2"
            >
              <span
                key="1"
              >
                the documentation on GitHub
              </span>
            </a>
            <span
              key="3"
            >
              .
            </span>
          </span>
        </p>
      </PanelBody>
    </Panel>
  </div>
</SideEffect(DocumentTitle)>
`;
